crates/proto/src/xfer/serial_message.rs

```diff
@@ -43,3 +43,15 @@ impl SerialMessage {
         Message::from_vec(&self.message)
     }
 }
+
+impl From<(Vec<u8>, SocketAddr)> for SerialMessage {
+    fn from((message, addr): (Vec<u8>, SocketAddr)) -> Self {
+        SerialMessage { message, addr }
+    }
+}
+
+impl From<SerialMessage> for (Vec<u8>, SocketAddr) {
+    fn from(msg: SerialMessage) -> Self {
+        msg.into_parts()
+    }
+}
```

Comment:
I think I'd prefer inherent `into_parts()` and `from_parts()` for this, what do you think?

Reply:
If we're talking about my subjective opinion, I prefer using traits from stdlib where possible. It opens up the door for polymorphism like `I: Into<SerialMessage>` even outside of `trust-dns`. If you don't like it, I can backtrack this change though.

Reply:
I like your second suggestion. I find trait definitions on tuples to be a bit hidden and hard to find. I don't mind having them, but even as someone who's worked quite a bit with Rust, the tuple Into's and From's are never easy to see. But I like being more polymorphic with the stdlib, that's a good goal.

Reply:
Sorry I clicked the wrong emoji there and I have no idea how to remove it now, haha